apps/test/unit/templates/progress/SendLessonTest.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import {assert} from '../../../util/reconfiguredChai';
+import SendLesson from '@cdo/apps/templates/progress/SendLesson';
+import {shallow} from 'enzyme';
+
+const lessonUrl = 'https://studio.code.org/s/coursee-2020/stage/2/puzzle/1';
+const lessonTitle = 'Lesson 2: Drawing with Loops';
+
+describe('SendLesson', () => {
+  it('renders a button with a share icon', () => {
+    const wrapper = shallow(
+      <SendLesson lessonUrl={lessonUrl} lessonTitle={lessonTitle} />
+    );
+
+    assert.equal(wrapper.find('Button').length, 1);
+    assert.equal(
+      wrapper
+        .find('Button')
+        .at(0)
+        .props().icon,
+      'share-square-o'
+    );
+  });
+
+  it('opens the SendLessonDialog when the button is clicked', () => {
+    const wrapper = shallow(
+      <SendLesson lessonUrl={lessonUrl} lessonTitle={lessonTitle} />
+    );
+
+    // dialog should be closed initially
+    assert.equal(wrapper.find('Connect(SendLessonDialog)').length, 0);
+
+    // click the button
+    assert.equal(wrapper.find('Button').length, 1);
+    wrapper.find('Button').simulate('click');
```

Comment:
Rather than using simulate, it's better practice to call the function directly
`wrapper.instance().openDialog()`
Details: https://github.com/enzymejs/enzyme/issues/2173#issuecomment-505551552

Reply:
Oh interesting!  I actually kind of wanted to test that the click handler is hooked up properly (which calling openDialog directly doesn't do, right?).  Is there a better way to test that the click handler is hooked up properly?

Reply:
Ok, I'll give (1) a shot.  For my education, why is using simulate() discouraged?